workers/unity/Packages/com.improbable.gdk.buildsystem/WorkerBuilder.cs

```diff
@@ -115,7 +129,24 @@ public static void BuildWorkerForEnvironment(string workerType, BuildEnvironment
 
             foreach (var unityBuildTarget in GetUnityBuildTargets(buildPlatforms))
             {
-                BuildWorkerForTarget(workerType, unityBuildTarget, buildOptions, targetEnvironment);
+                var buildTargetGroup = BuildPipeline.GetBuildTargetGroup(unityBuildTarget);
+                var activeScriptingBackend = PlayerSettings.GetScriptingBackend(buildTargetGroup);
+                try
+                {
+                    if (scriptingBackend != null)
+                    {
+                        Debug.Log($"Setting scripting backend to {scriptingBackend.Value}");
+                        PlayerSettings.SetScriptingBackend(buildTargetGroup, scriptingBackend.Value);
+                    }
+                    
+                    BuildWorkerForTarget(workerType, unityBuildTarget, buildOptions, targetEnvironment);
+                }
+                catch (Exception e)
+                {
+                    PlayerSettings.SetScriptingBackend(buildTargetGroup, activeScriptingBackend);
```

Comment:
Should we move this to finally block? Seems a bit odd that we only restore scripting backend if we throw

Reply:
good spot, that was meant to be in the finally part